src/main/java/ch/njol/skript/SkriptCommand.java

```diff
@@ -497,20 +587,128 @@ private static List<File> getSubFiles(File file) {
 		return files;
 	}
 
-	private static @Nullable File getScriptFromArgs(CommandSender sender, String[] args) {
-		return getScriptFromArgs(sender, args, Skript.getInstance().getScriptsFolder());
+	private static File @Nullable [] getScriptsFromArgs(CommandSender sender, String[] args) {
+		return getScriptsFromArgs(sender, args, Skript.getInstance().getScriptsFolder());
+	}
+
+	private static File @Nullable [] getScriptsFromArgs(CommandSender sender, String[] args, File directoryFile) {
+		boolean enable = args[0].equals("enable");
+		Map<String, File> scripts = new HashMap<>();
+		for (String currentScript : parseAsCommaSeparatedList(false, true, args)) {
+			File thisScript = getScriptFromArg(sender, currentScript, directoryFile);
+			if (thisScript == null) {
+				// Always allow '/' and '\' regardless of OS
+				boolean directory = currentScript.endsWith("/") || currentScript.endsWith("\\") || currentScript.endsWith(File.separator);
+				Skript.error(sender, (directory ? m_invalid_folder : m_invalid_script).toString(currentScript));
+				continue;
+			}
+			scripts.put(currentScript, thisScript);
+		}
+
+		Map<String, File> filteredScripts = new HashMap<>();
+		Set<Entry<String, File>> entries = scripts.entrySet();
+		// Now we check to see if any of the files found is already set to be loaded if the directory was also provided
+		for (Entry<String, File> entry : entries) {
```

Comment:
I think this whole set of for loops could be replaced with a stream and a reduce() call